AppHandling/Run-AlPipeline.ps1

```diff
@@ -2559,6 +2646,98 @@ if ($buildArtifactFolder -and (Test-Path $bcptResultsFile)) {
+
+${env:containerUsername} = $credential.UserName
+${env:containerPassword} = $credential.Password | Get-PlainText
+$startAddress = "http://$containerName/BC?tenant=default"
+$usedNames = @()
+
+$pageScriptingTests | ForEach-Object {
+    $thisFailed = $false
+    if ($restoreDatabases -contains 'BeforeEachPageScriptingTest') {
+        Write-GroupStart -Message "Restoring databases before each page scripting test"
+        Invoke-Command -ScriptBlock $RestoreDatabasesInBcContainer -ArgumentList @{"containerName" = $containerName}
+        Write-GroupEnd
+    }
+    $testSpec = $_
+    $name = $testSpec -replace '[\\/]', '-' -replace ':', '' -replace '\*', 'all' -replace '\?', 'x' -replace '\.yml$', ''
+    if ($usedNames -contains $name) {
+        throw "PageScriptingTests contains two similar test specs (resulting in identical results folders), please rename your test specs ($testSpec)."
+    }
+    $usedNames += $name
+    $path = $testSpec
+    if (-not [System.IO.Path]::IsPathRooted($path)) { $path = Join-Path $baseFolder $path }
+    if (-not (Test-Path $path)) { throw "No page scripting tests found matching $testSpec" }
+    Write-Host "Running Page Scripting Tests for $testSpec (test name: $name)"
+    $resultsFolder = Join-Path $pageScriptingTestResultsFolder $name
+    New-Item -Path $resultsFolder -ItemType Directory | Out-Null
+    pwsh -command {
+        npx replay $args[0] -ResultDir $args[1] -StartAddress $args[2] -Authentication UserPassword -usernameKey 'containerUsername' -passwordkey 'containerPassword'
+    } -args $path, $resultsFolder, $startAddress
+    if ($? -ne "True") {
+        Write-Host "Page Scripting Tests failed for $testSpec"
+        $allPassed = $false
+        $thisFailed = $true
+    }
+    $testResultsFile = Join-Path $resultsFolder "results.xml"
+    $playwrightReportFolder = Join-Path $resultsFolder 'playwright-report'
+    if ((Test-Path $testResultsFile -PathType Leaf) -and (Test-Path $playwrightReportFolder -PathType Container)) {
+        $thisXml = [xml](Get-Content $testResultsFile -encoding UTF8)
+        $thisXml.testsuites.testsuite.Name = $name
+        $resultsXml = $thisXml
+        if (Test-Path $pageScriptingTestResultsFile) {
+            # Merge results and aggregate counts
+            $resultsXml = [xml](Get-Content $pageScriptingTestResultsFile -encoding UTF8)
+            $resultsXml.testsuites.AppendChild($resultsXml.ImportNode($thisXml.testsuites.testsuite, $true))
+        }
+        foreach($property in 'tests','failures','skipped','errors','time') {
+            $resultsXml.testsuites."$property" = "$(([double[]]$resultsXml.testsuites.testsuite."$property" | Measure-Object -Sum).Sum)"
+        }
+        $resultsXml.Save($pageScriptingTestResultsFile)
+        Remove-Item $testResultsFile -Force
+        if ($thisFailed) {
+            Write-Host "Moving Playwright report folder"
+            Move-Item -Path "$playwrightReportFolder/*" -Destination $resultsFolder -Force

```

Comment:
Won't this override the existing playwright-report folder?

Reply:
This moves a subfolder one up - else the results will be two subfolders down.
Other results will be in a different resultsfolder (with the name of the page scripting test)